### PR TITLE
[AI] Rework phantom guest ai

### PR DESCRIPTION
**Pull request by miraco**
Some fixes for https://github.com/cmangos/issues/issues/3323
Not sure the spell_template change belongs here (maybe into core repo?)

Maybe timers need to get some tweaks (i took them from wotlk classic sniff and videos)

### CODE REVIEW

File: Updates/0707_phantom_guest.sql [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+DELETE FROM creature_spell_list_entry WHERE Id IN (1640901, 1640902, 1640903, 1640904, 1640905);
+REPLACE INTO `creature_spell_list_entry` (Id, Name, ChanceSupportAction, ChanceRangedAttack) VALUES
+(1640901, "Karazhan - Phantom Guest (Engineer)", 0, 0),
+(1640902, "Karazhan - Phantom Guest (Healer)", 0, 0),
+(1640903, "Karazhan - Phantom Guest (Hunter)", 0, 75),
+(1640904, "Karazhan - Phantom Guest (Caster)", 0, 75),
+(1640905, "Karazhan - Phantom Guest (Melee)", 0, 0);
+
+-- Dance Vibe spell should not stack 
+UPDATE `spell_template` SET `StackAmount`='1' WHERE (`Id`='29521');
```
Comment (AnonXS) [inline Updates/0707_phantom_guest.sql:29]: this has to be pred against core, or if allowed, we could place it as custom fix in db.

**Comment by AnonXS** [comment]
It's still being spammed.

grafik

File: ACID/acid_tbc.sql [hunk 2]
```diff
@@ -28068,23 +28068,18 @@ INSERT INTO `creature_ai_scripts` (`id`,`creature_id`,`event_type`,`event_invers
...
-('1640916','16409','9','123','100','1025','0','30','2400','3800','0','0','11','29579','1','256','0','0','0','0','0','0','0','0','Phantom Guest - Cast Throw Dynamite (Phase 2)'),
-('1640917','16409','9','119','100','1025','5','30','2300','3900','0','0','11','29582','1','256','0','0','0','0','0','0','0','0','Phantom Guest - Cast Throw (Phase 3)'),
+('1640901','16409','11','0','100','0','0','0','0','0','0','0','31','1','5','0','0','0','0','0','0','0','0','0','Phantom Guest - Set Random Phase 1-5 on Spawn'),
+('1640902','16409','4','0','10','0','0','0','0','0','0','0','1','13838','13839','13840','0','0','0','0','0','0','0','0','Phantom Guest - Say on Aggro'),
+('1640903','16409','6','0','10','0','0','0','0','0','0','0','1','13841','13842','13843','0','0','0','0','0','0','0','0','Phantom Guest - Say on Death'),
+('1640904','16409','4','25','100','0','0','0','0','0','0','0','57','0','0','0','5','0','0','0','47','0','0','0','Phantom Guest - Enable Melee Mode and Remove Emote and Set Stand State Stand on Aggro (Phase 1,2,5)'),
+('1640905','16409','4','61','100','0','0','0','0','0','0','0','60','1640901','0','0','0','0','0','0','0','0','0','0','Phantom Guest - Set SpellList 01 (Phase 1'),
```
Comment (AnonXS) [inline ACID/acid_tbc.sql:28075]: ) here and there missing

**Comment by miraco** [comment]
> It's still being spammed.
> 
> grafik
> 
> As this is all a regression we might not be able to fix it db side.
> 
> I blame something with SPELL_EFFECT_APPLY_AREA_AURA_PARTY going wrong when checking for if aura is present @killerwife
> 
> I also get
> 
> ```
> EventAI: Creature entry 16409 has ranged mode action but no main spell.
> ```
> 
> because main spell is in creature_spell_lists and not in EAI.
> 
> the aura is the issue still, being removed and recast, it seems it sets them into combat and they instantly evade afterwards recasting the aura promptly.
> 
> /edit: Leaving this open for now due to issues, merged with. - [7796dfe](https://github.com/cmangos/tbc-db/commit/7796dfe4c00b56360e2e86851ac155c3c060a657)

i finaly had sometime to test these things again to see why there are some bugs.

First one:
yes its getting spammed because the npcs around the one who actualy casts the spell on his self getting the buff removed/added all the time, so yes we cant fix it db side. its a bit better explained in here https://github.com/cmangos/issues/issues/3323
the ai is indeed correct, we need to use 27 - EVENT_T_MISSING_AURA 

For the other error (EventAI: Creature entry 16409 has ranged mode action but no main spell.) i dont get why tbh but i think its something with creature_ai_scripts ACTION_T_SET_SPELL_SET not working correctly.

if you manualy add one of the range spell_list to creature_template.SpellList it works as it should.

**Pull request closed without merging**